Sources/Testing/Traits/AttachmentSavingTrait.swift

```diff
@@ -0,0 +1,337 @@
+
+/// A type that defines a condition which must be satisfied for the testing
+/// library to save attachments recorded by a test.
+///
+/// To add this trait to a test, use one of the following functions:
+///
+/// - ``Trait/savingAttachments(if:)``
+///
+/// By default, the testing library saves your attachments as soon as you call
+/// ``Attachment/record(_:named:sourceLocation:)``. You can access saved
+/// attachments after your tests finish running:
+///
+/// - When using Xcode, you can access attachments from the test report.
+/// - When using Visual Studio Code, the testing library saves attachments to
+///   `.build/attachments` by default. Visual Studio Code reports the paths to
+///   individual attachments in its Tests Results panel.
+/// - When using Swift Package Manager's `swift test` command, you can pass the
+///   `--attachments-path` option. The testing library saves attachments to the
+///   specified directory.
+///
+/// If you add an instance of this trait type to a test, any attachments that
+/// test records are stored in memory until the test finishes running. The
+/// testing library then evaluates the instance's condition and, if the
+/// condition is met, saves the attachments.
+@_spi(Experimental)
+public struct AttachmentSavingTrait: TestTrait, SuiteTrait {
+  /// A type that describes the conditions under which the testing library
+  /// will save attachments.
+  ///
+  /// You can pass instances of this type to ``Trait/savingAttachments(if:)``.
+  public struct Condition: Sendable {
+    /// The testing library saves attachments if the test passes.
+    public static var testPasses: Self {
+      Self { !$0.hasFailed }
+    }
+
+    /// The testing library saves attachments if the test fails.
+    public static var testFails: Self {
+      Self { $0.hasFailed }
+    }
+
+    /// The testing library saves attachments if the test records a matching
+    /// issue.
+    ///
+    /// - Parameters:
+    ///   - issueMatcher: A function to invoke when an issue occurs that is used
+    ///     to determine if the testing library should save attachments for the
+    ///     current test.
+    ///
+    /// - Returns: An instance of ``AttachmentSavingTrait/Condition`` that
+    ///   evaluates `issueMatcher`.
+    public static func testRecordsIssue(
+      matching issueMatcher: @escaping @Sendable (_ issue: Issue) async throws -> Bool
+    ) -> Self {
+      Self(inspectsIssues: true) { context in
+        for issue in context.issues {
+          if try await issueMatcher(issue) {
+            return true
+          }
+        }
+        return false
+      }
+    }
+
+    /// Whether or not this condition needs to inspect individual issues (which
+    /// implies a slower path.)
+    fileprivate var inspectsIssues = false
+
+    /// The condition function.
+    ///
+    /// - Parameters:
+    ///   - condition: The function to call. The result of this function
+    ///     determines if the condition is satisfied or not.
```

Comment:
Do the docs for this closure property need to be updated?

Reply:
Clarify?

Reply:
It's `borrowing Context`, I must have dropped the label. Will re-add it.

Reply:
Eh, removed the comment.